src/main/java/seedu/address/logic/commands/StatisticsCommand.java

```diff
@@ -18,38 +18,78 @@ public class StatisticsCommand extends Command {
         }
+    }
+
+    private String buildStatisticsMessage(HashMap<JobCode, JobCodeStatistics> jobStatisticsMap) {
+        StringBuilder statisticsMessage = new StringBuilder();
+        appendHeading(statisticsMessage, "STATISTICS");
+        appendHeading(statisticsMessage, "Total number of applicants by Job Code and Interview Stages");
+
+        int totalApplicantsInSystem = 0;
+
+        // Iterate over the job statistics and build the message
+        for (JobCode jobCode : jobStatisticsMap.keySet()) {
+            JobCodeStatistics jobStats = jobStatisticsMap.get(jobCode);
+
+            appendJobCodeStatistics(statisticsMessage, jobCode.value, jobStats.getTotalApplicants());
 
-        StringBuilder statisticsMessage = new StringBuilder("STATISTICS\n"
-                + "--------\n"
-                + "Total number of applicants: " + numOfApplicants + "\n"
-                + "Number of applicants by Job Code\n"
-                + "--------\n");
-
-        // Append JobCode statistics to the message
-        for (JobCode jobCode : jobDict.keySet()) {
-            statisticsMessage.append(jobCode.value)
-                    .append(": ")
-                    .append(jobDict.get(jobCode))
-                    .append("\n");
+            // Appends breakdown of interview stages of job applicants
+            appendStagesBreakdown(statisticsMessage, jobStats);
+
+            // Update the total count of applicants in the system
+            totalApplicantsInSystem += jobStats.getTotalApplicants();
         }
-        return new CommandResult(statisticsMessage.toString());
+
+        // Total number of applicants
+        statisticsMessage.append("\nTotal number of applicants in the system: ")
+                .append(totalApplicantsInSystem)
+                .append("\n");
+
+        return statisticsMessage.toString();
+    }
+
+    private void appendJobCodeStatistics(StringBuilder statisticsMessage, String jobCode, int totalApplicants) {
+        statisticsMessage.append(jobCode)
+                .append(": ")
+                .append(totalApplicants)
+                .append(" applicants\n");
+    }
+
+    // Helper method to append formatted headings
+    private void appendHeading(StringBuilder builder, String heading) {
+        builder.append(heading).append("\n")
+                .append("--------\n");
+    }
+
+    // New method to append stages breakdown
+    private void appendStagesBreakdown(StringBuilder builder, JobCodeStatistics jobStats) {
+        builder.append("      N: ").append(jobStats.getN())
+                .append("      TP: ").append(jobStats.getTP())
+                .append("      TC: ").append(jobStats.getTC())
+                .append("      BP: ").append(jobStats.getBP())
+                .append("      BC: ").append(jobStats.getBC())
+                .append("      A: ").append(jobStats.getA())
+                .append("      R: ").append(jobStats.getR()).append("\n");
```

Comment:
Maybe it might be better to use SIX_SPACES constant instead of manually putting the spaces for "      N: "